l10n_ch_scan_bvr/wizard/scan_bvr.py

```diff
@@ -485,61 +348,64 @@ def validate_bvr_string(self, cr, uid, ids, context):
+            partners_bank = partner_bank_model.search(domain)
+            if len(partners_bank) == 1:
+                if (
+                        not partners_bank[0].bvr_adherent_num or
+                        partners_bank[0].bvr_adherent_num ==
+                        data['bvr_struct']['bvrnumber']):
+                    partner_bank = partners_bank
+            elif len(partners_bank) > 1:
+                # We need to filter further by bvr_adherent_num
+                partners_bank = partners_bank.filtered(
+                    lambda r:
+                    r.bvr_adherent_num == data['bvr_struct']['bvrnumber'])
+                if len(partners_bank) > 1:
+                    raise UserError(
+                        _('There are more than one bank corresponding '
+                          'to the current string.\n'
+                          'Please check the banks configuration.'))
+                partner_bank = partners_bank
         # We will need to know if we need to create invoice line
-        if partner_bank_search:
+        if partner_bank:
             # We have found the account corresponding to the
             # bvr_adhreent_number
             # so we can directly create the account
-            partner_bank_result = self.pool.get('res.partner.bank').browse(
-                cr,
-                uid,
-                partner_bank_search[0],
-                context=context
-            )
-            data['id'] = bvr_data.id
-            data['partner_id'] = partner_bank_result.partner_id.id
-            data['bank_account'] = partner_bank_result.id
-            data['journal_id'] = bvr_data.journal_id.id
-            action = self._create_direct_invoice(cr, uid, ids, data, context)
+            data['id'] = self.id
+            data['partner_id'] = partner_bank.partner_id.id
+            data['bank_account'] = partner_bank.id
+            data['journal_id'] = self.journal_id.id
+            action = self._create_direct_invoice(data)
             return action
-        elif bvr_data.bank_account_id:
-            data['id'] = bvr_data.id
-            data['partner_id'] = bvr_data.partner_id.id
-            data['journal_id'] = bvr_data.journal_id.id
-            data['bank_account'] = bvr_data.bank_account_id.id
-            action = self._create_direct_invoice(cr, uid, ids, data, context)
+        elif self.bank_account_id:
+            data['id'] = self.id
+            data['partner_id'] = self.partner_id.id
+            data['journal_id'] = self.journal_id.id
+            data['bank_account'] = self.bank_account_id.id
+            action = self._create_direct_invoice(data)
             return action
         else:
             # we haven't found a valid bvr_adherent_number
             # we will need to create or update a bank account
-            self.write(cr, uid, ids, {'state': 'need_extra_info'})
+            self.write({'state': 'need_extra_info'})
```

Comment:
As we are in a method with `ensure_one` you can do `self.state = 'need_extra_info'`